components/badge/badge.component.ts

```diff
@@ -41,6 +44,8 @@ const NZ_CONFIG_MODULE_NAME: NzConfigKey = 'badge';
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
   animations: [zoomBadgeMotion],
+  standalone: true,
+  imports: [BidiModule, NgIf, NgStyle, NzBadgeSupComponent, NzOutletModule],
```

Comment:
We don't need to import BidiModule, it is optional for the component library and should be imported by the user. 

```ts
@Optional() private directionality: Directionality
```

WDYT?

https://ng.ant.design/docs/direction/en